File: basics/sort/selectsort.py
def selectsort(arr):
    arr_sorted=[]
    for i in range(len(arr)):
        arr_sorted.append(min(arr[i:]))
    return arr_sorted

if __name__ == "__main__":
    arr_test = [5,4,3,2,1]
    arr_sorted = selectsort(arr_test)
    print(arr_sorted)

File: cpp/basics/sort/selectsort.cpp
#include <iostream>

class selectSort {
public:
    void sort(int arr[], int size) {
        for(int i=0; i<size; i++) {
            int min = i;
            for(int j=i+1; j<size; j++) {
                if(arr[j] < arr[min]) {
                    min = j;
                }
            }
            std::swap(arr[i], arr[min]);
        }
    }
};

int main() {
    int nums[] = {1, 3, 2, 5, 4};
    selectSort s;
    s.sort(nums, 5);
    for(int i=0; i<5; i++) {
        std::cout<<nums[i]<<" ";
    }
    std::cout<<std::endl;
    return 0;
}

